### PR TITLE
(Hot-)Fixes for Krypton

### PR DESCRIPTION
**Pull request by manuelm**
Adds support for connection state change callback to fix connecting and reconnecting if the backend goes down. Plus removes RTSP, which isn't working anymore.

Edit: Will upmerge this afterwards

### CODE REVIEW

**Comment by manuelm** [comment]
Also fixes #45

File: src/DvbData.cpp [hunk 1]
```diff
@@ -590,66 +580,95 @@ const std::string &Dvb::GetLiveStreamURL(const PVR_CHANNEL &channelinfo)
...
+        XBMC->Log(LOG_INFO, "Connection to the backend service failed."
+          " Retrying in 10 seconds...");
+        Sleep(10000);
+      }
+    }
+    else
     {
-      m_updateTimers = false;
-      m_mutex.Unlock();
       Sleep(1000);
```
Comment (CiNcH83) [inline src/DvbData.cpp]: Please revisit the mutex locking. It always includes this Sleep which does not sound like a very good idea, does it?

File: src/DvbData.cpp [hunk 2]
```diff
@@ -590,66 +580,95 @@ const std::string &Dvb::GetLiveStreamURL(const PVR_CHANNEL &channelinfo)
...
+      {
+        m_updateEPG = false;
+        m_mutex.Unlock();
+        Sleep(8000); /* Sleep enough time to let the recording service grab the EPG data */
+        m_mutex.Lock();
+        XBMC->Log(LOG_INFO, "Performing forced EPG update!");
+        PVR->TriggerEpgUpdate(m_currentChannel);
+      }
+
+      if (m_updateTimers)
```
Comment (CiNcH83) [inline src/DvbData.cpp]: Do we really have to update the timers every second?

File: src/DvbData.cpp [hunk 3]
```diff
@@ -590,66 +580,95 @@ const std::string &Dvb::GetLiveStreamURL(const PVR_CHANNEL &channelinfo)
...
-      update = 0;
-      XBMC->Log(LOG_INFO, "Performing timer/recording updates!");
-      TimerUpdates();
-      PVR->TriggerRecordingUpdate();
+      CLockObject lock(m_mutex);
+      if (m_updateEPG)
+      {
+        m_updateEPG = false;
+        m_mutex.Unlock();
+        Sleep(8000); /* Sleep enough time to let the recording service grab the EPG data */
```
Comment (CiNcH83) [inline src/DvbData.cpp]: Is this sleep really necessary? The loop runs every second anyway!?

File: src/DvbData.cpp (hunk 1 above)
Comment (manuelm) [inline src/DvbData.cpp]: Your comment is referring to removed code. Better add/pin your comments to the new code.
And yes, that's intentional, although I have to admin I don't like this update loop-approach very much. The loop is still from the origin maintainer, I just made it more readable.

**Pull request merged**